airflow/api_connexion/openapi/v1.yaml

```diff
@@ -1381,11 +1381,198 @@ paths:
+          format: datetime
+          description: The date user was created
+          readOnly: true
+        changed_on:
+          type: string
+          format: datetime
+          description: The date user was changed
+          readOnly: true
+
+    UserCollection:
+      description: User collection
+      type: object
+      properties:
+        users:
+          type: array
+          items:
+          $ref: '#/components/schemas/UserCollectionItem'
+
+    UserLogin:
+      description: Login item
+      allOf:
+        - $ref: '#/components/schemas/UserCollectionItem'
+        - type: object
+          properties:
+            token:
+              type: string
+              nullable: false
+              description: JWT token
+
+    RoleCollectionItem:
+      description: Role collection item
+      type: object
+      properties:
+        id:
+          type: string
+          description: The role ID
+        name:
+          type: string
+          description: The name of the role
+        permissions:
+          type: array
+          items:
+            $ref: '#/components/schemas/PermissionView'
+
+    RoleCollection:
+      description: Role Collections
+      type: object
+      properties:
+        roles:
+          type: array
+          items:
+            $ref: '#/components/schemas/RoleCollectionItem'
+
+    PermissionCollectionItem:
+      description: Permission Collection Item
+      type: object
+      properties:
+        id:
+          type: string
+          description: The permission ID
+        name:
+          type: string
+          description: The name of the permission
+          nullable: false
+
+    PermissionCollection:
+      description: Permission Collection
+      type: object
+      properties:
+        permissions:
+          type: array
+          items:
+            $ref: '#/components/schemas/PermissionCollectionItem'
+
+    PermissionView:
+      description: Permission view item
+      type: object
+      properties:
+        id:
+          type: string
+          description: The PermissionView ID
+        permission_id:
+          type: string
+          description: The permission ID
+        permission:
+          type: string
+          description: The name of the permission
+        view_menu_id:
+          type: string
+          description: The view menu id
+        view_menu_name:
+          type: string
+          description: The view menu name
+
+    ViewMenuCollectionItem:
+      description: ViewMenu Collection Item
+      type: object
+      properties:
+        id:
+          type: string
+          description: The ViewMenu ID
+        name:
+          type: string
+          description: The name of the ViewMenu
+          nullable: false
+
+    ViewMenuCollection:
+      description: ViewMenu Collection
+      type: object
+      properties:
+        view_menus:
+          type: array
+          items:
+            $ref: '#/components/schemas/ViewMenuCollectionItem'
```

Comment:
@ryanahamilton @jhtimmins Does this unduly tie is un to using FABs permissions model forever?

Do we perhaps want to re-think this and choose what/how we expose permissions without just blindly copying FAB?

Reply:
@ashb Structurally, I think FAB has a good model. Roles are made up of discrete permissions, and users can have multiple roles. It's a common design and I don't have a problem with us using it indefinitely.

I don't think it makes sense to use FAB names though. `ViewMenu` and `Permission` really only makes sense in a FAB UI-centric world. I've switched (somewhat inconsistently, I'll admit) to using `Resource` instead of `ViewMenu` and `Action` instead of `Permission`. With the new naming, a `Permission = Resource + Action`.

Reply:
In the long run, I think it will make sense to get away from FAB and Flask altogether. To do that we can abstract away the security side of FAB into its own module (so yeah we'll basically have to copy/redo parts of FAB), then we can use the custom security module to whatever new framework we use. The existing architecture and updated names should give a solid base for this transition.

Reply:
How can we model this in a schema @jhtimmins

Reply:
@ephraimbuddy I think in the schema we can just change `ViewMenu` to `Resource` and `Permission` to `Action`. I can make a PR to update the documentation so we use the new names.

Reply:
@jhtimmins what do you think about roles coming out like this:
```
"roles": [
            {
                "id": 1,
                "name": "Admin",
                "permissions": [
                    {
                        "action": {
                            "id": 3,
                            "name": "can_delete"
                        },
                        "id": 1,
                        "resource": {
                            "id": 1,
                            "name": "Connections"
                        }
                    },
                    {
                        "action": {
                            "id": 1,
                            "name": "can_read"
                        },
                        "id": 2,
                        "resource": {
                            "id": 1,
                            "name": "Connections"
                        }
                    },
                  ]
            }
         ]

Reply:
@ephraimbuddy This looks great!